persistent/Database/Persist/Sql/Internal.hs

```diff
@@ -81,7 +81,7 @@ mkColumns
     -> BackendSpecificOverrides
     -> ([Column], [UniqueDef], [ForeignDef])
 mkColumns allDefs t overrides =
-    (cols, getEntityUniques t, getEntityForeignDefs t)
+    (cols, getEntityUniquesNoPrimaryKey t, getEntityForeignDefs t)
```

Comment:
In order to preserve migration compatibility, we don't return the composite primary key in the `UniqueDef` for migrations.